RecoEgamma/PhotonIdentification/plugins/PhotonMVAEstimatorRun2Spring15NonTrig.cc

```diff
@@ -0,0 +1,304 @@
+  
+  _tmvaReaders.clear();
+}
+
+void PhotonMVAEstimatorRun2Spring15NonTrig::setConsumes(edm::ConsumesCollector&& cc){
+
+  // All tokens for event content needed by this MVA
+  // Cluster shapes
+  _full5x5SigmaIEtaIEtaMapToken = cc.consumes <edm::ValueMap<float> >
+    (_conf.getParameter<edm::InputTag>("full5x5SigmaIEtaIEtaMap"));
+
+  _full5x5SigmaIEtaIPhiMapToken = cc.consumes <edm::ValueMap<float> >
+    (_conf.getParameter<edm::InputTag>("full5x5SigmaIEtaIPhiMap"));
+
+  _full5x5E1x3MapToken = cc.consumes <edm::ValueMap<float> >
+    (_conf.getParameter<edm::InputTag>("full5x5E1x3Map"));
+
+  _full5x5E2x2MapToken = cc.consumes <edm::ValueMap<float> >
+    (_conf.getParameter<edm::InputTag>("full5x5E2x2Map"));
+
+  _full5x5E2x5MaxMapToken = cc.consumes <edm::ValueMap<float> >
+    (_conf.getParameter<edm::InputTag>("full5x5E2x5MaxMap"));
+
+  _full5x5E5x5MapToken = cc.consumes <edm::ValueMap<float> >
+    (_conf.getParameter<edm::InputTag>("full5x5E5x5Map"));
+
+  _esEffSigmaRRMapToken = cc.consumes <edm::ValueMap<float> >
+    (_conf.getParameter<edm::InputTag>("esEffSigmaRRMap"));
+
+  // Isolations
+  _phoChargedIsolationToken = cc.consumes <edm::ValueMap<float> >
+    (_conf.getParameter<edm::InputTag>("phoChargedIsolation"));
+
+  _phoPhotonIsolationToken = cc.consumes <edm::ValueMap<float> >
+    (_conf.getParameter<edm::InputTag>("phoPhotonIsolation"));
+
+  _phoWorstChargedIsolationToken = cc.consumes <edm::ValueMap<float> >
+    (_conf.getParameter<edm::InputTag>("phoWorstChargedIsolation"));
+
+  // Pileup
+  _rhoToken = cc.consumes<double> (_conf.getParameter<edm::InputTag>("rho"));
+  
+}
+
+void PhotonMVAEstimatorRun2Spring15NonTrig::getEventContent(const edm::Event& iEvent){
+
+  // Get the full5x5 and ES maps
+  iEvent.getByToken(_full5x5SigmaIEtaIEtaMapToken, _full5x5SigmaIEtaIEtaMap);
+  iEvent.getByToken(_full5x5SigmaIEtaIPhiMapToken, _full5x5SigmaIEtaIPhiMap);
+  iEvent.getByToken(_full5x5E1x3MapToken, _full5x5E1x3Map);
+  iEvent.getByToken(_full5x5E2x2MapToken, _full5x5E2x2Map);
+  iEvent.getByToken(_full5x5E2x5MaxMapToken, _full5x5E2x5MaxMap);
+  iEvent.getByToken(_full5x5E5x5MapToken, _full5x5E5x5Map);
+  iEvent.getByToken(_esEffSigmaRRMapToken, _esEffSigmaRRMap);
+
+  // Get the isolation maps
+  iEvent.getByToken(_phoChargedIsolationToken, _phoChargedIsolationMap);
+  iEvent.getByToken(_phoPhotonIsolationToken, _phoPhotonIsolationMap);
+  iEvent.getByToken(_phoWorstChargedIsolationToken, _phoWorstChargedIsolationMap);
+
+  // Get rho
+  iEvent.getByToken(_rhoToken,_rho);
+
+  // Make sure everything is retrieved successfully
+  if(! (_full5x5SigmaIEtaIEtaMap.isValid()
+	|| _full5x5SigmaIEtaIPhiMap.isValid()
+	|| _full5x5E1x3Map.isValid()
+	|| _full5x5E2x2Map.isValid()
+	|| _full5x5E2x5MaxMap.isValid()
+	|| _full5x5E5x5Map.isValid()
+	|| _esEffSigmaRRMap.isValid()
+	|| _phoChargedIsolationMap.isValid()
+	|| _phoPhotonIsolationMap.isValid()
+	|| _phoWorstChargedIsolationMap.isValid()
+	|| _rho.isValid() ) )
```

Comment:
the logic ("at least something is retrieved")  doesn't match the comment "make sure everything is retrieved";